compiler/rustc_lint/src/impl_trait_overcaptures.rs

```diff
@@ -303,20 +303,28 @@ impl<'tcx> TypeVisitor<TyCtxt<'tcx>> for VisitOpaqueTypes<'tcx> {
                             ResolvedArg::EarlyBound(def_id) | ResolvedArg::LateBound(_, _, def_id),
                         ) => {
                             if self.tcx.def_kind(self.tcx.parent(def_id)) == DefKind::OpaqueTy {
-                                let (ty::ReEarlyParam(ty::EarlyParamRegion { def_id, .. })
-                                | ty::ReLateParam(ty::LateParamRegion {
-                                    bound_region: ty::BoundRegionKind::BrNamed(def_id, _),
-                                    ..
-                                })) = self
+                                let def_id = match self
                                     .tcx
                                     .map_opaque_lifetime_to_parent_lifetime(def_id.expect_local())
                                     .kind()
-                                else {
-                                    span_bug!(
+                                {
```

Comment:
Could this perhaps also use `Region::opt_param_def_id`?

like 
```
tcx.map_opaque_blah_blah_blah(def_id.expect_local()).opt_param_def_id().unwrap_or_else(|| {
    span_bug!(
        self.tcx.def_span(def_id),
        "variable should have been duplicated from a parent"
    )
})
```

Reply:
oh... this should have been in the type system commit whoops.